controller/backup_target_controller.go

```diff
@@ -329,25 +371,46 @@ func (btc *BackupTargetController) reconcile(name string) (err error) {
 		if reflect.DeepEqual(existingBackupTarget.Status, backupTarget.Status) {
 			return
 		}
-		if _, err := btc.ds.UpdateBackupTargetStatus(backupTarget); err != nil && apierrors.IsConflict(errors.Cause(err)) {
+		updatedBackupTarget, err := btc.ds.UpdateBackupTargetStatus(backupTarget)
+		if err != nil {
+			if apierrors.IsConflict(errors.Cause(err)) {
+				log.WithError(err).Debugf("Requeue %v due to conflict", name)
+				btc.enqueueBackupTarget(backupTarget)
+			}
+			if !apierrors.IsNotFound(err) {
+				log.WithError(err).Debugf("Failed to updated backup target %v", name)
+				return
+			}
+		}
+		if !backupTarget.DeletionTimestamp.IsZero() || updatedBackupTarget.Spec.Default == backupTarget.Spec.Default {
+			err = nil
+			return
+		}
+		updatedBackupTarget.Spec.Default = backupTarget.Spec.Default
```

Comment:
If status remains the same, the this spec change will never be applied.
Case:
1. Controller successfully updates Status but fails to update Spec in this iteration
2. In the next iteration, Controller will never reach this logic since we return if we status is the same [L371](https://github.com/longhorn/longhorn-manager/pull/2182/files#diff-b881cb9d9886c8b07c3081dc101c18685f066469dee34f80bcdb7b98f76acacdR371)

Reply:
Modified. Set a default backup target and disable the previous default backup target before we start to synchronize the backup target status.
Thanks